src/lib/flow/sol-flow-builder.c

```diff
@@ -544,6 +616,15 @@ get_type_description(struct sol_flow_builder *builder)
         builder->type_desc.ports_out = builder->ports_out_desc.base.data;
     }
 
+    if (builder->options_description.len > 0) {
+        struct sol_flow_node_options_member_description *sentinel;
+        sentinel = sol_vector_append(&builder->options_description);
+        SOL_NULL_CHECK(sentinel, NULL);
+        memset(sentinel, 0, sizeof(*sentinel));
+        builder->type_desc.options = get_options_description(builder);
+        SOL_NULL_CHECK(builder->type_desc.options, NULL);
```

Comment:
Yes, if this fails, everything about this builder instance will fail and the whole vector will be freed.